### ISSUE TITLE
[Camera Plugin] CameraPreview blurred and stretched on Android API 24

### ISSUE THREAD

**Issue by EArminjon** [flutter/flutter#56760]
## Steps to Reproduce
I use camera: ^0.5.7+4 with emulator Android 7 APi 24 800x1280:mdpi.

```dart
import 'package:flutter/material.dart';
import 'package:flutter/widgets.dart';
import 'package:camera/camera.dart';

class PicturePage extends StatefulWidget {

  @override
  State<StatefulWidget> createState() => _PicturePage();
}

class _PicturePage extends State<PicturePage> {
  List<CameraDescription> cameras;
  int index = 0;
  CameraController controller;

  @override
  void initState() {
    super.initState();

    init();
  }

  @override
  void dispose() {
    controller?.dispose();
    super.dispose();
  }

  init() async {
    cameras = await availableCameras();
    await initThisCamera(0);
  }

  initThisCamera(int id) async {
    controller = CameraController(cameras[id], ResolutionPreset.max, enableAudio: false);
    index = id;
    await controller.initialize();
    if (!mounted) return;
    setState(() {});
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      extendBodyBehindAppBar: true,
      backgroundColor: Theme.of(context).backgroundColor,
      body: body(),
    );
  }

  Widget body() {
    if (controller == null || !controller.value.isInitialized) return Center(child: CircularProgressIndicator());
    return Stack(
      children: [
        CameraPreview(controller),
      ],
    );
  }
}

```

**Expected results:** 

On api 28 : 
![Screenshot_1589015578](https://user-images.githubusercontent.com/37028599/81469610-2061b000-91e6-11ea-9913-a5c84d27fc56.png)

**Actual results:** 
On API 24 : 
![Screenshot_1589015564](https://user-images.githubusercontent.com/37028599/81469615-28b9eb00-91e6-11ea-9918-9ef199167f8e.png)

API 24 / 28 render same logs expected for API 24 where we can see  "set take picture size to 320x240"... What  ?

<details>
  <summary>Logs</summary>

```
[ +666 ms] I/RequestQueue( 7655): Repeating capture request cancelled.
[  +22 ms] I/CameraManager( 7655): Using legacy camera HAL.
[  +10 ms] I/CameraDeviceState( 7655): Legacy camera service transitioning to state CONFIGURING
[        ] I/RequestThread-0( 7655): Configure outputs: 2 surfaces configured.
[        ] D/Camera  ( 7655): app passed NULL surface
[        ] I/RequestThread-0( 7655): configureOutputs - set take picture size to 320x240
[        ] D/        ( 7655): HostConnection::get() New Host Connection established 0x84f7aac0, tid 7748
[        ] D/EGL_emulation( 7655): eglCreateContext: 0x851698e0: maj 2 min 0 rcv 2
[  +23 ms] D/EGL_emulation( 7655): eglMakeCurrent: 0x851698e0: ver 2 0 (tinfo 0x8c8d4e30)
[   +2 ms] E/EGL_emulation( 7655): eglQueryContext 32c0  EGL_BAD_ATTRIBUTE
[        ] E/EGL_emulation( 7655): tid 7748: eglQueryContext(1804): error 0x3004 (EGL_BAD_ATTRIBUTE)
[        ] I/CameraDeviceState( 7655): Legacy camera service transitioning to state IDLE
[        ] I/RequestQueue( 7655): Repeating capture request set.
[        ] W/LegacyRequestMapper( 7655): convertRequestMetadata - control.awbRegions setting is not supported, ignoring value
[        ] W/LegacyRequestMapper( 7655): convertRequestToMetadata - Ignoring android.lens.focusDistance false, only 0.0f is supported
[ +199 ms] I/CameraDeviceState( 7655): Legacy camera service transitioning to state CAPTURING
[  +71 ms] Sending to VM service: ext.flutter.platformOverride({isolateId: isolates/2875484225354463})
[        ] Sending to VM service: ext.flutter.platformOverride({value: android, isolateId: isolates/2875484225354463})
```

```
flutter doctor -v
[✓] Flutter (Channel stable, v1.17.0, on Linux, locale fr_FR.UTF-8)
    • Flutter version 1.17.0 at /home/student/flutter
    • Framework revision e6b34c2b5c (il y a 7 jours), 2020-05-02 11:39:18 -0700
    • Engine revision 540786dd51
    • Dart version 2.8.1

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /home/student/Android/Sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at:
      /home/student/Documents/Softwares/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Android Studio (version 3.6)
    • Android Studio at /home/student/Documents/Softwares/android-studio
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] IntelliJ IDEA Ultimate Edition (version 2019.3)
    • IntelliJ at /home/student/Documents/Softwares/idea-IU-193.6911.18
    • Flutter plugin version 45.0.2
    • Dart plugin version 193.6911.31

[✓] VS Code (version 1.44.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.9.1

[✓] Connected device (2 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 7.0 (API
      24) (emulator)
    • AOSP on IA Emulator       • emulator-5556 • android-x86 • Android 9 (API
      28) (emulator)

• No issues found!
```
</details>

**Comment by TahaTesser** [comment]
Issue reproduces, looks preview is being stretch, same cup is different size

### API 29
![Screenshot_1589187354](https://user-images.githubusercontent.com/48603081/81545029-ef27e200-9395-11ea-9c46-7fae26a3291c.png)

### API 24
![Screenshot_1589188215](https://user-images.githubusercontent.com/48603081/81545074-fea72b00-9395-11ea-90c7-4c846575745b.png)

<details> 
<summary>Code Sample</summary> 

``` 
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// ignore_for_file: public_member_api_docs

import 'dart:async';
import 'dart:io';

import 'package:camera/camera.dart';
import 'package:flutter/material.dart';
import 'package:path_provider/path_provider.dart';
import 'package:video_player/video_player.dart';

import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Material App',
      theme: ThemeData.dark(),
      home: PicturePage(),
    );
  }
}

class PicturePage extends StatefulWidget {
  @override
  State<StatefulWidget> createState() => _PicturePage();
}

class _PicturePage extends State<PicturePage> {
  List<CameraDescription> cameras;
  int index = 0;
  CameraController controller;

  @override
  void initState() {
    super.initState();

    init();
  }

  @override
  void dispose() {
    controller?.dispose();
    super.dispose();
  }

  init() async {
    cameras = await availableCameras();
    await initThisCamera(0);
  }

  initThisCamera(int id) async {
    controller =
        CameraController(cameras[id], ResolutionPreset.max, enableAudio: false);
    index = id;
    await controller.initialize();
    if (!mounted) return;
    setState(() {});
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      extendBodyBehindAppBar: true,
      backgroundColor: Theme.of(context).backgroundColor,
      body: body(),
    );
  }

  Widget body() {
    if (controller == null || !controller.value.isInitialized)
      return Center(child: CircularProgressIndicator());
    return Stack(
      children: [
        CameraPreview(controller),
      ],
    );
  }
}

``` 

</details> 

<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, 1.19.0-0.0.pre, on Mac OS X 10.15.4 19E287, locale
    en-GB)
    • Flutter version 1.19.0-0.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision a849daf283 (3 days ago), 2020-05-07 18:59:02 -0700
    • Engine revision 3953c3ccd1
    • Dart version 2.9.0 (build 2.9.0-5.0.dev 4da5b40fb6)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/tahatesser/Code/SDK
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.45.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.1

[✓] Connected device (3 available)
    • macOS      • macOS      • darwin-x64     • Mac OS X 10.15.4 19E287
    • Web Server • web-server • web-javascript • Flutter Tools
    • Chrome     • chrome     • web-javascript • Google Chrome 81.0.4044.138

• No issues found!
tahatesser@Tahas-Ma
``` 

</details>

**Comment by TahaTesser** [comment]
Using `camera: ^0.6.4+5`, still blurry API 24
<details> 
<summary>flutter doctor -v</summary> 

```bash
    • Flutter version 1.22.5 at C:\Code\flutter_stable
    • Framework revision 7891006299 (5 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Code\SDK
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[!] Android Studio (version 4.1.0)
    • Android Studio at C:\Code\android-studio
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.52.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.18.1

[✓] Connected device (1 available)
    • IN2011 (mobile) • c9d8ee0c • android-arm64 • Android 11 (API 30)

! Doctor found issues in 1 category.
PS C:\Users\Taha\AndroidStudioProjects> flutters doctor -v
[✓] Flutter (Channel stable, 1.22.5, on Microsoft Windows [Version 10.0.19042.685], locale en-US)
    • Flutter version 1.22.5 at C:\Code\flutter_stable
    • Framework revision 7891006299 (5 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Code\SDK
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[!] Android Studio (version 4.1.0)
    • Android Studio at C:\Code\android-studio
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.52.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.18.1

[✓] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android 7.0 (API 24) (emulator)

! Doctor found issues in 1 category.
```

```bash
    • Flutter version 1.22.5 at C:\Code\flutter_stable
    • Framework revision 7891006299 (5 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Code\SDK
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[!] Android Studio (version 4.1.0)
    • Android Studio at C:\Code\android-studio
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.52.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.18.1

[✓] Connected device (1 available)
    • IN2011 (mobile) • c9d8ee0c • android-arm64 • Android 11 (API 30)

! Doctor found issues in 1 category.
PS C:\Users\Taha\AndroidStudioProjects> flutters doctor -v
    • Flutter version 1.22.5 at C:\Code\flutter_stable
    • Framework revision 7891006299 (5 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Code\SDK
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[!] Android Studio (version 4.1.0)
    • Android Studio at C:\Code\android-studio
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.52.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.18.1

[✓] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android 7.0 (API 24) (emulator)

! Doctor found issues in 1 category.
PS C:\Users\Taha\AndroidStudioProjects> flutterm doctor -v
[✓] Flutter (Channel master, 1.26.0-2.0.pre.338, on Microsoft Windows [Version 10.0.19042.685], locale en-US)
    • Flutter version 1.26.0-2.0.pre.338 at C:\Code\flutter_master
    • Framework revision 16e63cf34f (3 hours ago), 2021-01-14 07:34:02 -0500
    • Engine revision 1b5e16cda2
    • Dart version 2.12.0 (build 2.12.0-223.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Code\SDK
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.8.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.8.30804.86
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.52.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 7.0 (API 24) (emulator)
    • Windows (desktop)                  • windows       • windows-x64    • Microsoft Windows [Version 10.0.19042.685]
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 87.0.4280.141
    • Edge (web)                         • edge          • web-javascript • Microsoft Edge 84.0.522.52

```

</details>

**Comment by ViniciusSossela** [comment]
Using ```camera: ^0.6.4+5```, blurry on Samsung S2 API 30.

**Comment by exaby73** [comment]
I can no longer reproduce this issue on `Master (3.10.0-7.0.pre.46)`. Therefore I will close this issue. If you can still reproduce it, please share a [complete and minimal, reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) and steps to reproduce the issue, along with the output of `flutter doctor -v`. Thank you.

<details><summary><code>flutter doctor -v</code> (Master)</summary>

```
[!] Flutter (Channel master, 3.10.0-7.0.pre.46, on macOS 13.3.1 22E261 darwin-arm64, locale en-US)
    • Flutter version 3.10.0-7.0.pre.46 on channel master at /Users/nabeelparkar/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.19.5/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nabeelparkar/fvm/versions/master. Consider adding /Users/nabeelparkar/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 59230759be (22 minutes ago), 2023-04-19 02:39:50 -0400
    • Engine revision 609f9d5364
    • Dart version 3.1.0 (build 3.1.0-15.0.dev)
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/nabeelparkar/Library/Android/sdk/
    • Platform android-33, build-tools 33.0.2
    • ANDROID_SDK_ROOT = /Users/nabeelparkar/Library/Android/sdk/
    • Java binary at: /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/222.4459.24.2221.9862592/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[✓] Android Studio (version 2022.1)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9619390/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/222.4459.24.2221.9862592/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.77.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.3.1 22E261 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Brave Browser 112.1.50.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>
